### ISSUE TITLE
[GMap] Track gets drawn over and over again

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#14032]
### Describe your problem!

Track gets drawn over and over again and cannot be removed anymore. I guess that something with the layer management goes wrong.

I already looked into it, but couldn't find the cause. Could you take a look @eddiemuc?

Beginning | after some time
-|-
![Videoframe_20230225_202613_com huawei himovie overseas](https://user-images.githubusercontent.com/64581222/221376034-33893ab4-b515-4e36-8d39-c8479d04598b.jpg)|![Videoframe_20230225_202619_com huawei himovie overseas](https://user-images.githubusercontent.com/64581222/221376037-8b5e7fca-92d8-4e15-bad3-9037fa813f43.jpg)

### How to reproduce?

- open gmap
- import gpx track
- set color to something transparent
- move map around and wait

### Actual result after these steps?

Color should stay as is

### Expected result after these steps?

There are artifacts which cannot be removed anymore. You can even mix colors when changing the track color after some time ;-)

### Reproducible

Yes

### c:geo Version

NB25.02.2023

### Additional Information

Maybe the cause for @alan666notb observation:

> A minor issue, possibly related, is that if you delete the route, it remains on the map until you close the map and re-open it.

_Originally posted by @alan666notb in https://github.com/cgeo/cgeo/issues/14028#issuecomment-1445162263_

**Comment by eddiemuc** [comment]
Is this a new issue or was it already in c:geo for some time?

I looked into Google maps layer management some weeks ago and noticed that it is very messed up. I can totally confirm that it tends to leave unremovable objects in the map after some time, especially if many objects/tracks are used. This is a problem in the core design and not easily fixable. (Short explanation: the asynchronous map editing tends to "forget" objects on the map when they are removed and then added again , which e.g. happens when routes are recalculated)

Question is: did something happen in the past to let this occur more often than before?

**Comment by moving-bits** [comment]
Not sure if it's related to this issue, but something has changed recently with drawing elements, see #13993, which has been reported by users for release 2023.02.12 only.

**Comment by eddiemuc** [comment]
Is this also only for GMao?

**Comment by moving-bits** [comment]
Both users have provided screenshots for Google Maps Satellite only. I'll ask both whether OSM is affected as well. (No reports for OSM so far on support.)

(_Originally posted by @moving-bits in https://github.com/cgeo/cgeo/issues/13993#issuecomment-1445198382_. Will track further insights on this in the related issue)

**Comment by eddiemuc** [comment]
I tried and agree that it can be easily provoked that old tracks remain on google (for example by quickly turning visibility of routes/tracks on and off over and over again). As mentioned above, this is explainable to me by existing asynchronous update mechanism.

My question remains: is this a new  behaviour? Or is it just turning up now because people are using the route/track features more frequently? 

(As you know I work on a completely new unified geoitem layer design in parallel, my intention is to solve the problem above there too)

**Comment by moving-bits** [comment]
I can't remember having tickets on support mentioning the track thing before, and haven't experienced it myself (but I'm rarely using GM). The drawing of multiple destination lines is definitively new, though.

**Comment by fm-sys** [comment]
I'm almost sure this issue is new. And sounds closely related to https://github.com/cgeo/cgeo/issues/13993 / #14068

**Comment by MagpieFourtyTwo** [comment]
@eddiemuc 
> Is this a new issue or was it already in c:geo for some time?

Maybe no real help, as I don't know how you define "for some time", but at least that in GMap&GSat routing lines appear to be way more prominent and with higher opaqueness than in OSM (i. e. are most probably drawn multiple times w/o deleting the old one) is already there (in the nightlies) for at least several weeks. 

I can remember my astonishment, when I realized it for the first time (is a while ago), but can not remember the exact time nor the other changes that came with it.  Moreover I did not pay too much attention to it, as I don't use GMap at all and GSat just rarely and typically in "static" situations, to get a picture of cache surroundings and road situations (would never ever have even thought about navigating with GSat - ridiculous ;) ).

About a week ago I shortly noticed that there are even multiple routing lines on switch of routing mode, but did not stumble upon it till yesterday (see #14068). Looks like this may have the same cause - and is more easy to reproduce thus hopefully better to debug.

**Comment by MagpieFourtyTwo** [comment]
Seems to be solved, too, with today's nightly (resp. [5a5cfe1](https://github.com/cgeo/cgeo/commit/5a5cfe11d72fe3dc3a67c6f169b9d60238412ddc) resp. #14083) - at least the routing lines now have the expected opaqueness again and they do not get more prominent over time (at least over here, within about a quarter of an hour).

**Comment by fm-sys** [comment]
Yes, can confirm that it is fixed
